src/components/text-input/index.md

```diff
@@ -168,6 +168,16 @@ You will not normally need to use the `autocomplete` attribute in prototypes, as
 
 Users often need to copy and paste information into a text input, so do not stop them from doing this.
 
+### Avoid restricting the length of a user's input
+
+Using the `maxlength` attribute can confuse users as it doesn't provide feedback that the text has been truncated, especially in cases where the text has been copied and pasted from elsewhere. This can cause users to accidentally provide incorrect information.
```

Comment:
```suggestion
Using the `maxlength` attribute means there is no feedback to users that their text input is truncated. This is especially true where the text has been copied and pasted from elsewhere. This can cause users to accidentally provide incorrect or incomplete information.
```